### PR TITLE
Add option to suppress warning about the Microsoft C/C++ extension incompatibility

### PR DESCRIPTION
**Pull request by matms**
I've added an option to control suppression of the incompatibility warning (and not change the C/C++ settings).

I'm making this pull request as suggested in https://github.com/clangd/vscode-clangd/issues/162, since others seem to have faced this same issue.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -148,6 +148,11 @@
                         "Automatically restart the server",
                         "Do nothing"
                     ]
+                },
+                "clangd.detectAndDisableExtensionConflicts": {
```
Comment (kirillbobyrev) [inline package.json:152]: nit: how about `clangd.detectExtensionConflicts` + `Warn about conflicting extensions and suggest disabling them.`?

File: src/extension.ts [hunk 2]
```diff
@@ -27,23 +27,28 @@ export async function activate(context: vscode.ExtensionContext) {
...
+      if (cppTools && cppTools.isActive) {
+        const cppToolsConfiguration = vscode.workspace.getConfiguration('C_Cpp');
+        const cppToolsEnabled = cppToolsConfiguration.get('intelliSenseEngine');
+        if (cppToolsEnabled !== 'Disabled') {
+          vscode.window
+              .showWarningMessage(
+                  'You have both the Microsoft C++ (cpptools) extension and ' +
+                      'clangd extension enabled. The Microsoft IntelliSense features ' +
+                      'conflict with clangd\'s code completion, diagnostics etc.',
+                  'Disable IntelliSense')
```
Comment (kirillbobyrev) [inline src/extension.ts:45]: Nit: I think the perfect UX/UI would be adding a button `Don't show this warning`.

Something like 

```typescript
vscode.window.showWarningMessage('', ['Disable IntelliSense', 'Don\'t show this warning']).
    then(selection => {
        if (selection = 'Don\'t show this warning') {
            /* update clangd.detectAndDisableExtensionConflicts similar to what we
               have with C_Cpp.intelliSenseEngine right now */
        }
    }
```

**Review by kirillbobyrev — CHANGES_REQUESTED** [review]
Hey, sorry, I originally missed the patch. Thanks for picking it up and working on this! The logic looks good, I have a couple of suggestions that would make this patch even more awesome, would be happy to accept once resolved!

**Comment by kirillbobyrev** [comment]
(also, please `clang-format` the code before submitting, otherwise the CI linters will complain)

**Comment by matms** [comment]
Done. This patch also means you don't have to restart VsCode, since it should now clear the interval upon pressing 'Never show this message'.

Note that if this option is later used for _other_ incompatibilities, then pressing 'Never show this message' would disable all of them.

**Comment by kirillbobyrev** [comment]
Tested, works great, thank you for working on this!

**Pull request merged**